zerver/views/registration.py

```diff
@@ -155,6 +160,7 @@ def accounts_register(
     source_realm_id: Optional[int] = REQ(
         default=None, converter=to_converted_or_fallback(to_non_negative_int, None)
     ),
+    is_demo_organization: Optional[bool] = REQ(converter=to_bool, default=False),
```

Comment:
I think you want to use `validator=check_bool`, not `converter`; at least that's what essentially all of our other endpoints do.  And then we don't need to add a new duplicate validation function.

Reply:
We actually can't use `validator=check_bool` here. I am pretty sure `check_bool` is only used as `json_validator=check_bool`. The request for creating a realm is not a JSON request. If you look at [the POST request we make for realm creation](https://github.com/zulip/zulip/blob/main/zerver/views/development/registration.py#L103), the booleans `terms` and `is_demo_organization` are passed as strings `"true"`. When you change the above code to use `validator=check_bool`, `mypy` complains with the following:

```
mypy      | zerver/views/registration.py:164: error: No overload variant of "REQ" matches argument types "Callable[[str, object], bool]", "bool"  [call-overload]
mypy      | zerver/views/registration.py:164: note: Possible overload variants:
mypy      | zerver/views/registration.py:164: note:     def [ResultT] REQ(whence: Optional[str] = ..., *, converter: Callable[[str], ResultT], default: ResultT = ..., intentionally_undocumented: bool = ..., documentation_pending: bool = ..., aliases: Sequence[str] = ..., path_only: bool = ...) -> ResultT
mypy      | zerver/views/registration.py:164: note:     def [ResultT] REQ(whence: Optional[str] = ..., *, default: ResultT = ..., json_validator: Callable[[str, object], ResultT], intentionally_undocumented: bool = ..., documentation_pending: bool = ..., aliases: Sequence[str] = ..., path_only: bool = ...) -> ResultT
mypy      | zerver/views/registration.py:164: note:     def REQ(whence: Optional[str] = ..., *, default: str = ..., str_validator: Optional[Callable[[str, object], str]] = ..., intentionally_undocumented: bool = ..., documentation_pending: bool = ..., aliases: Sequence[str] = ..., path_only: bool = ...) -> str
mypy      | zerver/views/registration.py:164: note:     def REQ(whence: Optional[str] = ..., *, default: None, str_validator: Optional[Callable[[str, object], str]] = ..., intentionally_undocumented: bool = ..., documentation_pending: bool = ..., aliases: Sequence[str] = ..., path_only: bool = ...) -> Optional[str]
mypy      | zerver/views/registration.py:164: note:     def [ResultT] REQ(whence: Optional[str] = ..., *, default: ResultT = ..., argument_type: Literal['body'], intentionally_undocumented: bool = ..., documentation_pending: bool = ..., aliases: Sequence[str] = ..., path_only: bool = ...) -> ResultT
mypy      |
mypy      | See https://zulip.readthedocs.io/en/latest/testing/mypy.html for debugging tips.
```

So the only way to convert these strings into a `bool` was to add a dedicated converter here. It could be that I am missing something here, let me know what you think! Thanks! :)

Reply:
Hmm.  I don't we want to keep the `validator.py` library as a pure library for things that match the `json_validator` system.  @andersk do you have a recommendation for how we should validate a boolean for this function?